source/client-side-operations-timeout/client-side-operations-timeout.rst

```diff
@@ -0,0 +1,927 @@
+
+Retryability
+~~~~~~~~~~~~
+
+If an operation requires a retry per the retryable reads or writes
+specifications and ``timeoutMS`` is set to a non-zero value, drivers MUST
+retry operations as many times as possible before the timeout expires or a
+retry attempt returns a non-retryable error. Once the timeout expires, a
+timeout error MUST be raised.
+
+See `Why don’t drivers use backoff/jitter between retry attempts?`_.
+
+Client Side Encryption
+~~~~~~~~~~~~~~~~~~~~~~
+
+If automatic client-side encryption or decryption is enabled, the remaining
+``timeoutMS`` value MUST be used as the ``timeoutMS`` when executing
+``listCollections`` commands to retrieve collection schemas, ``find``
+commands to get data from the key vault, and any commands against
+mongocryptd. It MUST also be used as the request timeout for HTTP requests
+against KMS servers to decrypt data keys. When sending a command to
+mongocryptd, drivers MUST NOT append a ``maxTimeMS`` field. This is to ensure
+that a ``maxTimeMS`` field can be safely appended to the command after it has
+been marked by mongocryptd and encrypted by libmongocrypt. To determine
+whether or not the server is a mongocryptd, drivers MUST check that the
+``iscryptd`` field in the server's description is ``true``.
+
+For explicit encryption and decryption, the ``ClientEncryptionOpts`` options
+type used to construct `ClientEncryption
+<../client-side-encryption/client-side-encryption.rst#clientencryption>`_
+instances MUST support a new ``timeoutMS`` option, which specifies the timeout
+for all operations executed on the ``ClientEncryption`` object.
+
+See `maxTimeMS is not added for mongocryptd`_.
+
+Background Connection Pooling
+-----------------------------
+
+Connections created as part of a connection pool’s ``minPoolSize``
+maintenance routine MUST use ``connectTimeoutMS`` as the timeout for
+connection establishment. After the connection is established, if
+``timeoutMS`` is set at the MongoClient level, it MUST be used as the timeout
+for all commands sent as part of the MongoDB or authentication handshakes.
+The timeout MUST be refreshed after each command. These commands MUST set
+timeouts per the `Command Execution`_ section. If ``timeoutMS`` is not set,
+drivers MUST continue to honor ``socketTimeoutMS`` as the socket timeout for
+handshake and authentication commands.
+
+Server Monitoring
+-----------------
+
+Drivers MUST NOT use ``timeoutMS`` for commands executed by the server
+monitoring and RTT calculation threads.
+
+See `Monitoring threads do not use timeoutMS`_.
+
+Cursors
+-------
+
+For operations that create cursors, ``timeoutMS`` can either cap the lifetime
+of the cursor or be applied separately to the original operation and all
+``next`` calls. To support both of these use cases, these operations MUST
+support a ``timeoutMode`` option. This option is an enum with possible values
+``CURSOR_LIFETIME`` and ``ITERATION``. The default value depends on the type
```

Comment:
Did we talk about changing `CURSOR_LIFETIME` to just `LIFETIME` so that the enum could have a more generic name and ultimately apply to GridFS down the line (instead of having to introduce `STREAM_LIFETIME`)?

Even without that, an argument could be made that "CURSOR_" is redundant since it would very well be a prefix for "ITERATION" as well.